core/src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala

```diff
@@ -118,7 +118,7 @@ class JdbcAsyncWriteJournal(config: Config) extends AsyncWriteJournal {
     journalDao
       .messagesWithBatch(persistenceId, fromSequenceNr, toSequenceNr, journalConfig.daoConfig.replayBatchSize, None)
       .take(max)
-      .mapAsync(1)(reprAndOrdNr => Future.fromTry(reprAndOrdNr))
+      .collect { case Success(reprAndOrdNr) => reprAndOrdNr }
```

Comment:
This has different behaviour than the original logic though, that would fail the stream on a Failure while this filters it out. Better would be to skip this operator/step completely and do the match in `runForEach` and throw for Failure there:

```scala
      .runForeach { 
        case Success((repr, _)) =>
          recoveryCallback(repr)
        case Failure(ex) => throw ex
      }
```

Reply:
More effiency, Yay!

```
[info] Benchmark                                     Mode  Cnt       Score        Error  Units
[info] JdbcAsyncJournalBenchmark.originalAsync      thrpt    3  581405.864 ± 217516.676  ops/s
[info] JdbcAsyncJournalBenchmark.streamWithCollect  thrpt    3  594420.619 ±  29680.962  ops/s

[info] Benchmark                                     Mode  Cnt       Score       Error  Units
[info] JdbcAsyncJournalBenchmark.originalAsync      thrpt    3  577291.504 ± 51386.930  ops/s
[info] JdbcAsyncJournalBenchmark.streamWithCollect  thrpt    3  605493.214 ±  4152.262  ops/s
```

Reply:
Push my commit again, because i havn't permission to apply this request change,